homeassistant/components/hinen_power/__init__.py

```diff
@@ -0,0 +1,96 @@
+"""Support for hello auth."""
+
+from __future__ import annotations
+
+from aiohttp.client_exceptions import ClientError, ClientResponseError
+from openai import BaseModel
+
+from homeassistant.components.application_credentials import ClientCredential
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.config_entry_oauth2_flow import (
+    AbstractOAuth2Implementation,
+    OAuth2Session,
+    async_get_config_entry_implementation,
+    async_register_implementation,
+)
+
+from . import application_credentials
+from .auth_config import AsyncConfigEntryAuth
+from .const import AUTH, COORDINATOR, DOMAIN
+from .coordinator import HinenDataUpdateCoordinator
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: HinenIntegrationConfigEntry
+) -> bool:
+    """Set up the Hinen Auth component."""
+    hinen_auth_impl: AbstractOAuth2Implementation = (
+        await application_credentials.async_get_auth_implementation(
+            hass,
+            DOMAIN,
+            ClientCredential(
+                entry.data["token"]["client_id"], entry.data["token"]["client_secret"]
+            ),
+        )
+    )
+    async_register_implementation(hass, DOMAIN, hinen_auth_impl)
+    implementation = await async_get_config_entry_implementation(hass, entry)
+    auth = AsyncConfigEntryAuth(hass, OAuth2Session(hass, entry, implementation))
+
+    try:
+        await auth.check_and_refresh_token()
+    except ClientResponseError as err:
+        if 400 <= err.status < 500:
+            raise ConfigEntryAuthFailed(
+                "OAuth session is not valid, reauth required"
+            ) from err
+        raise ConfigEntryNotReady from err
+    except ClientError as err:
+        raise ConfigEntryNotReady from err
+
+    coordinator = HinenDataUpdateCoordinator(hass, entry, auth)
+    await coordinator.async_config_entry_first_refresh()
+    await delete_devices(hass, entry, coordinator)
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        COORDINATOR: coordinator,
+        AUTH: auth,
+    }
```

Comment:
Don't store the coordinator in `hass.data`, unless the coordinator is shared between config entries (that doesn't seem to be the case?). Instead, store it in `entry.runtime_data`